posts/creating-your-first-truffle-project.md

```diff
@@ -0,0 +1,634 @@
+    
+    You can randomly generate by running ganach-cli without one. For example `node_modules/.bin/ganache-cli | grep Mnemonic` will output the single line with it. Then you can just kill the process with ^C. ![Ganache cli mnemonic (pic missing)](https://raw.githubusercontent.com/tylerjohnhaden/blog-usa/master/images/2019/01/ganache_cli_mnemonic.png)
+
+4. Run Ganache and see what is generated
+
+        npm run ganache
+
+    ![Ganache cli (pic missing)](https://raw.githubusercontent.com/tylerjohnhaden/blog-usa/master/images/2019/01/ganache_cli.png)
+    Ganache will generate accounts based on what parameters you run it with. The default is 10 with starting balances of 100 Ether. The cli will display the addresses, private keys, mnemonic, gas price, and gas limit. These addresses can technically be used on any Ethereum blockchain, not just you local one (but they probably have 0 real Ether).
+    
+    This Ganache client will sit around, waiting for someone to send it a transaction on port 8545 by default. When it receives that transaction, it will attempt to run it on the EVM (see if the bytecode is correct) and it will then immediately create a single block with that transaction (mine the block) . On the main Ethereum blockchain, several transactions will be added to any given block, but we can be less efficient on our local version. All clients like this one should have a specific set of api calls that can read or write to the blockchain. This is why all transactions are public to everyone in the network.
+    
+5. Test the Ganache client by sending an api call from another terminal
+
+        curl http://127.0.0.1:8545 \
+            -X POST \
+            -H "Content-Type: application/json" \
+            -d '{"jsonrpc": "2.0", "method": "web3_clientVersion"}'
+
+    ![Ganache cli curl (pic missing)](https://raw.githubusercontent.com/tylerjohnhaden/blog-usa/master/images/2019/01/ganache_cli_curl.png)
+    ![Ganache cli log (pic missing)](https://raw.githubusercontent.com/tylerjohnhaden/blog-usa/master/images/2019/01/ganache_cli_log.png)
+    As you can see, our client responds with a client version, which tells us which protocol to use. Don't worry, you probably won't have to deal with different protocol versions if you are reading this article. This is just a way to test that your client is running properly.
+    
+    These api calls (like `web3_clientVersion`) are part of Ethereum's protocols. Most clients will support the majority of these methods. For an explanation of these methods see the [JSON RPC docs](https://github.com/ethereum/wiki/wiki/JSON-RPC). 
+
+
+## Deploy to your local blockchain
+
+Now that we have a blockchain client to store our transactions, lets deploy our contract from before to it. In Truffle lingo, we say that were going to migrate our project to our development network. Well need to add some more configurations and then it will be as simple as running a command.
```

Comment:
```suggestion
Now that we have a blockchain client to store our transactions, let's deploy our contract from before to it. In Truffle lingo, we say that we're going to migrate our project to our development network. We'll need to add some more configurations and then it will be as simple as running a command.
```